### PR TITLE
PS-392 use phr:dbx-subdefs-api

### PR DESCRIPTION
**Pull request by jygaulier**
replace phr api me/subdefs by databoxes/subdefs, including "type"

### CODE REVIEW

**Comment by jygaulier** [comment]
do not merge now

File: .env [hunk 1]
```diff
@@ -174,7 +174,7 @@ PS_GATEWAY_IP=172.30.0.1
 
 # Xdebug
 IDE_KEY=PHPSTORM
-XDEBUG_ENABLED=0
+XDEBUG_ENABLED=1
```
Comment (4rthem) [inline .env]: you should use a custom env var or set it in your `env.local`

**Review by 4rthem — CHANGES_REQUESTED** [review]

File: databox/indexer/src/handlers/phraseanet/phraseanetClient.ts [hunk 2]
```diff
@@ -71,17 +71,23 @@ export default class PhraseanetClient {
...
+        const dbxs = res.data.response.databoxes;
+        Object.keys(dbxs).forEach(id => {
+            const defs = dbxs[id].subdefs;
+            Object.keys(defs).forEach(type => {
+                const sd = defs[type];
+
+                Object.keys(sd).forEach(name => {
+                    let s = sd[name];
+                    s['type'] = type;   // could be included by api, but for now add it here
+                    subdefs.push(s);
```
[the remark this answers is not in the document]
Comment (jygaulier) [reply databox/indexer/src/handlers/phraseanet/phraseanetClient.ts:89]: ok i'll have a look on "..." syntax

File: databox/indexer/src/handlers/phraseanet/types.d.ts [hunk 3]
```diff
@@ -36,9 +36,11 @@ export type PhraseanetMetaStruct = {
 }
 
 export type PhraseanetSubDef = {
-    name: string;
+    type: string;                       // image | video | audio | document
```
[the remark this answers is not in the document]
Comment (jygaulier) [reply databox/indexer/src/handlers/phraseanet/types.d.ts:39]: ts can check at runtime ?

**Pull request merged**